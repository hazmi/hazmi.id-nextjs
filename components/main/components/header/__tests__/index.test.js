import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import Header from '../index'

const data = [
  {
    id: 1,
    title: "",
    link: "/",
    icon: <span />,
  },
  {
    id: 2,
    title: "Work Experiences",
    link: "/work-experiences",
    icon: <span />,
  }
];

describe('With Enzyme', () => {
  it('Show the site name', () => {
    const router = {
      pathname: '/'
    };
    const wrapper = shallow(<Header data={data} router={router} />)
    expect(wrapper.dive().find('h1').text()).toEqual('Hazmi.ID')
  })
  it('Show the correct breadcrumbs in title', () => {
    const router = {
      pathname: '/work-experiences'
    };
    const wrapper = shallow(<Header data={data} router={router} />)
    expect(wrapper.dive().find('h1').text()).toEqual('Hazmi.ID / Work Experiences')
  })
})

describe('With Snapshot Testing', () => {
  it('It show the site name', () => {
    const router = {
      pathname: '/'
    };
    const component = renderer.create(<Header data={data} router={router} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('It show the correct breadcrumbs', () => {
    const router = {
      pathname: '/work-experiences'
    };
    const component = renderer.create(<Header data={data} router={router} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})