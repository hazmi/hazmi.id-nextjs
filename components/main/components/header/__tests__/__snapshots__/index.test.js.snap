// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing It show the correct breadcrumbs 1`] = `
<div
  className="header"
>
  <img
    alt="Avatar Image"
    className="header__avatar"
    height={32}
    src="/static/pp.jpg"
    width={32}
  />
  <h1
    className="header__title"
  >
    Hazmi.ID / Work Experiences
  </h1>
</div>
`;

exports[`With Snapshot Testing It show the site name 1`] = `
<div
  className="header"
>
  <img
    alt="Avatar Image"
    className="header__avatar"
    height={32}
    src="/static/pp.jpg"
    width={32}
  />
  <h1
    className="header__title"
  >
    Hazmi.ID
  </h1>
</div>
`;
